iosApp/iosApp/ViewModels/SettingsViewModel.swift

```diff
@@ -94,9 +88,10 @@ class SettingsViewModel: ObservableObject {
         }
     }
 
-    private func setSettings(_ settings: [Setting]) {
+    private func setSettings(_ settings: [Settings: Bool]) {
         Task {
-            try await settingsRepository.setSettings(settings: Set(settings))
+            try await settingsRepository.setSettings(settings: settings.mapValues { KotlinBoolean(bool: $0) })
+            await self.getSections()
```

Comment:
Refreshing the sections here was necessary to get the toggle switches to update their state correctly - without it, the states were writing, but the new values weren't being read back. I'm not sure why that'd be a consequence of my changes.